rest-api-spec/src/main/resources/rest-api-spec/api/index.json

```diff
@@ -101,6 +101,10 @@
       "require_alias": {
         "type": "boolean",
         "description": "When true, requires destination to be an alias. Default is false"
+      },
+      "require_data_stream": {
+        "type": "boolean",
+        "description": "When true, requires destination to have a parent data stream. Default is false"
```

Comment:
Since index is passed on to the bulk action and the bulk action resolves the destination to a backing index before checking, should this state that the flag requires the target to be a data stream instead?